### PR TITLE
Fix rebase conflict detection in git 2.26

### LINKED ISSUE #10928: [Bug] Support git v2.26.0
git made some changes witch changed the way how rebase-merge info is stored ...

### PR DESCRIPTION
**Pull request by zeripath**
Git changed the technique used in rebase from simple apply-patches to use merge. This breaks our conflict detection code.

Fix #10928 

Signed-off-by: Andrew Thornton <art27@cantab.net>

### CODE REVIEW

**Review by 6543 — APPROVED** [review]

**Comment by 6543** [comment]
need backport

File: services/pull/merge.go [hunk 1]
```diff
@@ -263,17 +263,34 @@ func rawMerge(pr *models.PullRequest, doer *models.User, mergeStyle models.Merge
...
+				var commitSha string
+				if _, statErr := os.Stat(filepath.Join(tmpBasePath, ".git", "rebase-apply", "original-commit")); statErr == nil {
+					// The original commit SHA1 that is failing will be in .git/rebase-apply/original-commit
+					commitShaBytes, readErr := ioutil.ReadFile(filepath.Join(tmpBasePath, ".git", "rebase-apply", "original-commit"))
+					if readErr != nil {
+						// Abandon this attempt to handle the error
+						log.Error("git rebase staging on to base [%s:%s -> %s:%s]: %v\n%s\n%s", pr.HeadRepo.FullName(), pr.HeadBranch, pr.BaseRepo.FullName(), pr.BaseBranch, err, outbuf.String(), errbuf.String())
+						return "", fmt.Errorf("git rebase staging on to base [%s:%s -> %s:%s]: %v\n%s\n%s", pr.HeadRepo.FullName(), pr.HeadBranch, pr.BaseRepo.FullName(), pr.BaseBranch, err, outbuf.String(), errbuf.String())
+					}
+					commitSha = strings.TrimSpace(string(commitShaBytes))
```
Comment (techknowlogick) [inline services/pull/merge.go]: Looks great.

One very tiny nit, you have `commitSha = strings.TrimSpace(string(commitShaBytes))` twice, could this be taken out of the if conditionals and be move below before the `log.Debug`?

**Comment by 6543** [comment]
@zeripath please send second patch ;)

**Review by 6543 — APPROVED** [review]

**Comment by zeripath** [comment]
@6543 So I think the second commit was sent by the time you commented.

**Review by lafriks — APPROVED** [review]

**Pull request merged**